nodegsettings.cc

```diff
@@ -0,0 +1,385 @@
+/*
+ * GPII Node.js GSettings Bridge
+ *
+ * Copyright 2012 Steven Githens
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ *
+ */
+
+/*
+ * For changes done after initial fork from
+ * https://github.com/GPII/linux
+ * based on commit b28e99212e4c7cc803f210c066e0124d72c8b72d.
+ *
+ * Copyright 2015 Leopold Burdyl
+ */
+
+#include <node.h>
+#include <v8.h>
+#include <gio/gio.h>
+
+using namespace v8;
+
+// globals
+const GVariantType* X_G_VARIANT_TYPE_STRING_TUPLE_ARRAY = g_variant_type_new("a(ss)");
+
+
+/* Helper function - because we do this A LOT */
+char * v8_value_to_utf8_string(const Local<Value> value_obj) {
+	Local<String>  string_obj                = value_obj->ToString();
+	int            utf8_string_length        = string_obj->Utf8Length();
+	char          *utf8_string               = (char*) g_malloc(utf8_string_length + 1);
```

Comment:
yep. I fixed the memory leaks for g_malloc and some gio api calls that require a specific free call. I guess in the future it would be good to integrate some memory checks into the tests.